java/vector/src/main/java/org/apache/arrow/vector/complex/FixedSizeListVector.java

```diff
@@ -17,61 +17,44 @@
 
 package org.apache.arrow.vector.complex;
 
-import static java.util.Collections.singletonList;
-import static org.apache.arrow.vector.complex.BaseRepeatedValueVector.DATA_VECTOR_NAME;
-
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collections;
-import java.util.Iterator;
 import java.util.List;
-import java.util.Objects;
 
-import org.apache.arrow.memory.BaseAllocator;
 import org.apache.arrow.memory.BufferAllocator;
 import org.apache.arrow.memory.OutOfMemoryException;
 import org.apache.arrow.memory.util.ByteFunctionHelpers;
 import org.apache.arrow.util.Preconditions;
-import org.apache.arrow.vector.AddOrGetResult;
 import org.apache.arrow.vector.BaseValueVector;
 import org.apache.arrow.vector.BitVectorHelper;
-import org.apache.arrow.vector.BufferBacked;
 import org.apache.arrow.vector.FieldVector;
 import org.apache.arrow.vector.ValueVector;
 import org.apache.arrow.vector.ZeroVector;
 import org.apache.arrow.vector.complex.impl.UnionFixedSizeListReader;
 import org.apache.arrow.vector.complex.impl.UnionFixedSizeListWriter;
+import org.apache.arrow.vector.complex.impl.UnionListWriter;
 import org.apache.arrow.vector.ipc.message.ArrowFieldNode;
 import org.apache.arrow.vector.types.Types.MinorType;
 import org.apache.arrow.vector.types.pojo.ArrowType;
 import org.apache.arrow.vector.types.pojo.DictionaryEncoding;
-import org.apache.arrow.vector.types.pojo.Field;
 import org.apache.arrow.vector.types.pojo.FieldType;
 import org.apache.arrow.vector.util.CallBack;
 import org.apache.arrow.vector.util.JsonStringArrayList;
-import org.apache.arrow.vector.util.OversizedAllocationException;
-import org.apache.arrow.vector.util.SchemaChangeRuntimeException;
 import org.apache.arrow.vector.util.TransferPair;
 
 import io.netty.buffer.ArrowBuf;
 
 /** A ListVector where every list value is of the same size. */
-public class FixedSizeListVector extends BaseValueVector implements FieldVector, PromotableVector {
+public class FixedSizeListVector extends ListVector implements FieldVector, PromotableVector {
```

Comment:
I don't think this is correct.  BaseRepeatedValueVector, the parent of ListVector contains an offsetBuffer member.  We don't want to include that in FixedSizeListVector.  I think we should create an interface  that both BaseRepeatedVector/ListVector and FixedSizeListVector both inherit from sow they can be used interchangeably in dictionary encoding.

I also responded on the ML thread to see if other people have opinions.

Reply:
Sure, I just thought introduced an new interface makes the hierarchy a little confused. Let's hold up and see if there has other opinions, otherwise I will refactor as you suggested. Thanks!